test/tokengenerator.go

```diff
@@ -0,0 +1,141 @@
+package testutils
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+const (
+	bitSize = 2048
+)
+
+// WebKeySet represents a JWK Set object.
+type WebKeySet struct {
+	Keys []jwk.Key `json:"keys"`
+}
+
+// ExtraClaim a function to set claims in the token to generate
+type ExtraClaim func(token *jwt.Token)
+
+// WithEmailClaim sets the `email` claim in the token to generate
+func WithEmailClaim(email string) ExtraClaim {
+	return func(token *jwt.Token) {
+		token.Claims.(jwt.MapClaims)["email"] = email
+	}
+}
+
+// Identity is a user identity
+type Identity struct {
+	ID       uuid.UUID
+	Username string
+}
+
+// NewIdentity returns a new, random identity
+func NewIdentity() *Identity {
+	return &Identity{
+		ID:       uuid.NewV4(),
+		Username: "testuser-" + uuid.NewV4().String(),
+	}
+}
+
+// TokenGenerator represents the test token and key manager.
+type TokenGenerator struct {
```

Comment:
What about naming it `TokenManager` since it not only generates, but also stores keys?

Reply:
fixed here: 56af8a9f5f97c17b87ede58a884271c3691ea3b1